feat(content-server): Enable gzip for html and static resources on the content server

shane-tomlinson opened the pull request:
This should better approximate prod and ship ~ 1/2 the bytes down the wire.

@vladikoff, @jrgm or @jbuck - r?

Commit:
feat(content-server): Enable gzip for html and static resources on the content server

This should better approximate prod.

jrgm reviewed (changes requested):
Differences:
 - prod gzip_comp_level is 1, not 6
 - gzip_buffers - perhaps gzip_buffers 32 4k to match PAGESIZE
 - gzip_min_length 1100 vs. default 20
 - gzip_types
   - prod is:
      application/javascript;                                                                                                               
      application/json                                                                                                                      
      application/x-javascript                                                                                                              
      application/xml                                                                                                                       
      application/xml+rss                                                                                                                   
      text/css                                                                                                                              
      text/javascript                                                                                                                       
      text/plain                                                                                                                            
      text/xml                                                                                                                              
                                                                                                                                            
I think the gzip_comp_level and gzip_min_length make sense (90% of the compression in doing any compression at all; don't gzip if you won't save packets). On gzip_types, up to you, and if prod should change, let me know.

jrgm commented:
Also, prod has an includes/gzip.conf and does `include includes/gzip.conf`. Could do the same, and include it in each `server` block, but given the gzip_min_size, most api responses wouldn't be compressed anyways.

Commit:
fix(content-server): updates based on @jrgm's recommendations

* gzip_comp_level 1
* gzip_buffers 32 4k
* gzip_min_length 1100
* gzip_types application/javascript application/json application/x-javascript
  application/xml application/xml+rss text/css text/javascript text/plain text/xml

shane-tomlinson commented:
@jrgm - updated!
